codec/src/main/java/io/netty/handler/codec/compression/Lz4FrameEncoder.java

```diff
@@ -330,15 +391,24 @@ private ChannelHandlerContext ctx() {
     }
 
     @Override
-    public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+    public void handlerAdded(ChannelHandlerContext ctx) {
         this.ctx = ctx;
         // Ensure we use a heap based ByteBuf.
         buffer = Unpooled.wrappedBuffer(new byte[blockSize]);
+        buffer.clear();
```

Comment:
Yes, it is as `Unpooled.wrappedBuffer()` sets the `writeIndex` to the length of the array and we need the write index to be at 0 when we start using this class.